--- origsrc/libcxx-14.0.5.src/src/CMakeLists.txt	2022-06-10 22:47:36.000000000 +0800
+++ src/libcxx-14.0.5.src/src/CMakeLists.txt	2023-12-15 01:01:15.478232200 +0800
@@ -300,7 +300,7 @@
   cxx_add_common_build_flags(cxx_static)
   cxx_set_common_defines(cxx_static)
 
-  if (LIBCXXABI_USE_LLVM_UNWINDER)
+  if (LIBCXXABI_USE_LLVM_UNWINDER AND NOT LIBCXX_STANDALONE_BUILD)
     # If we intend to use the just-built unwinder, add a dependency so that it
     # gets built, even if we technically aren't going to link it in at this
     # stage.
