--- origsrc/libcxx-5.0.1.src/src/system_error.cpp	2017-05-31 17:14:05.000000000 -0500
+++ src/libcxx-5.0.1.src/src/system_error.cpp	2018-01-03 16:55:38.030845000 -0600
@@ -74,7 +74,7 @@ string do_strerror_r(int ev) {
   return string(buffer);
 }
 #elif defined(__linux__) && !defined(_LIBCPP_HAS_MUSL_LIBC) &&                 \
-    (!defined(__ANDROID__) || __ANDROID_API__ >= 23)
+    (!defined(__ANDROID__) || __ANDROID_API__ >= 23) || defined(__CYGWIN__)
 // GNU Extended version
 string do_strerror_r(int ev) {
     char buffer[strerror_buff_size];
