--- origsrc/libcxx-5.0.1.src/CMakeLists.txt	2017-07-19 09:04:38.000000000 -0500
+++ src/libcxx-5.0.1.src/CMakeLists.txt	2018-01-03 17:06:50.464926300 -0600
@@ -458,7 +458,9 @@ add_compile_flags_if_supported(-nostdinc
 # Hide all inline function definitions which have not explicitly been marked
 # visible. This prevents new definitions for inline functions from appearing in
 # the dylib when get ODR used by another function.
+if (NOT CYGWIN)
 add_compile_flags_if_supported(-fvisibility-inlines-hidden)
+endif()
 
 if (LIBCXX_CONFIGURE_IDE)
   # This simply allows IDE to process <experimental/coroutine>