--- origsrc/libcxx-5.0.1.src/include/__config	2017-07-17 16:52:31.000000000 -0500
+++ src/libcxx-5.0.1.src/include/__config	2018-01-03 16:55:38.020842500 -0600
@@ -43,7 +43,7 @@
 #define _LIBCPP_OBJECT_FORMAT_ELF   1
 #elif defined(__MACH__)
 #define _LIBCPP_OBJECT_FORMAT_MACHO 1
-#elif defined(_WIN32)
+#elif defined(_WIN32) || defined(__CYGWIN__)
 #define _LIBCPP_OBJECT_FORMAT_COFF  1
 #else
 #error Unknown object file format
@@ -252,6 +252,10 @@
 #endif
 #endif // defined(_WIN32)
 
+#ifdef __CYGWIN__
+#  define _LIBCPP_SHORT_WCHAR   1
+#endif
+
 #ifdef __sun__
 # include <sys/isa_defs.h>
 # ifdef _LITTLE_ENDIAN
@@ -887,7 +891,7 @@ template <unsigned> struct __static_asse
 
 #if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
 // Most unix variants have catopen.  These are the specific ones that don't.
-#if !defined(__BIONIC__) && !defined(_NEWLIB_VERSION)
+#if !defined(__BIONIC__) && !defined(_NEWLIB_VERSION) || defined(__CYGWIN__)
 #define _LIBCPP_HAS_CATOPEN 1
 #endif
 #endif
@@ -998,6 +1002,7 @@ _LIBCPP_FUNC_VIS extern "C" void __sanit
     defined(__APPLE__) || \
     defined(__CloudABI__) || \
     defined(__sun__) || \
+    defined(__CYGWIN__) || \
     (defined(__MINGW32__) && __libcpp_has_include(<pthread.h>))
 #   define _LIBCPP_HAS_THREAD_API_PTHREAD
 # elif defined(_LIBCPP_WIN32API)
@@ -1085,7 +1090,7 @@ _LIBCPP_FUNC_VIS extern "C" void __sanit
 #endif
 
 #if !defined(_LIBCPP_HAS_NO_OFF_T_FUNCTIONS)
-#if defined(_LIBCPP_MSVCRT) || defined(_NEWLIB_VERSION)
+#if defined(_LIBCPP_MSVCRT) || (defined(_NEWLIB_VERSION) && !defined(__CYGWIN__))
 #define _LIBCPP_HAS_NO_OFF_T_FUNCTIONS
 #endif
 #endif
