--- origsrc/libcxx-7.0.1.src/cmake/config-ix.cmake	2017-05-25 18:43:42.000000000 -0400
+++ src/libcxx-7.0.1.src/cmake/config-ix.cmake	2019-05-07 15:15:09.775006200 -0400
@@ -50,6 +50,17 @@ if (LIBCXX_SUPPORTS_NODEFAULTLIBS_FLAG)
                         shell32 user32 kernel32 mingw32 ${MINGW_RUNTIME}
                         moldname mingwex msvcrt)
     list(APPEND CMAKE_REQUIRED_LIBRARIES ${MINGW_LIBRARIES})
+  elseif (CYGWIN)
+    # Mingw64 requires quite a few "C" runtime libraries in order for basic
+    # programs to link successfully with -nodefaultlibs.
+    if (LIBCXX_USE_COMPILER_RT)
+      set(CYGWIN_RUNTIME ${LIBCXX_BUILTINS_LIBRARY} gcc_s)
+    else ()
+      set(CYGWIN_RUNTIME gcc_s gcc)
+    endif()
+    set(CYGWIN_LIBRARIES ${CYGWIN_RUNTIME} cygwin advapi32
+                        shell32 user32 kernel32 ${CYGWIN_RUNTIME})
+    list(APPEND CMAKE_REQUIRED_LIBRARIES ${CYGWIN_LIBRARIES})
   endif()
   if (CMAKE_C_FLAGS MATCHES -fsanitize OR CMAKE_CXX_FLAGS MATCHES -fsanitize)
     set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -fno-sanitize=all")
--- origsrc/libcxx-7.0.1.src/lib/CMakeLists.txt	2019-05-07 15:11:51.478191900 -0400
+++ src/libcxx-7.0.1.src/lib/CMakeLists.txt	2019-05-07 15:16:22.666699900 -0400
@@ -91,6 +91,7 @@ else()
   add_library_flags_if(LIBCXX_HAS_GCC_S_LIB gcc_s)
 endif()
 add_library_flags_if(LIBCXX_HAVE_CXX_ATOMICS_WITH_LIB atomic)
+add_library_flags_if(CYGWIN "${CYGWIN_LIBRARIES}")
 add_library_flags_if(MINGW "${MINGW_LIBRARIES}")
 
 # Add the unwinder library.
