--- origsrc/libcxx-7.0.1.src/include/__config	2018-08-17 03:22:08.000000000 -0400
+++ src/libcxx-7.0.1.src/include/__config	2019-05-07 15:08:09.181233200 -0400
@@ -1189,7 +1189,7 @@ _LIBCPP_FUNC_VIS extern "C" void __sanit
 #  endif
 #endif
 
-#if __has_attribute(require_constant_initialization)
+#if __has_attribute(require_constant_initialization) && !defined(__CYGWIN__) && !defined(__MINGW32__)
 #  define _LIBCPP_SAFE_STATIC __attribute__((__require_constant_initialization__))
 #else
 #  define _LIBCPP_SAFE_STATIC
